build/debian/tribler/usr/share/metainfo/org.tribler.Tribler.metainfo.xml

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>org.tribler.Tribler</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>LGPL-3.0-only</project_license>
+  <name>Tribler</name>
+  <developer_name>The Tribler Community</developer_name>
+  <url type="homepage">https://tribler.org/</url>
+  <url type="bugtracker">https://github.com/Tribler/tribler/issues</url>
+  <url type="help">https://github.com/Tribler/tribler/wiki</url>
+  <summary>Towards making Bittorrent anonymous and impossible to shut down.</summary>
+  <description>
+    <p>We use our own dedicated Tor-like network for anonymous torrent downloading.</p>
+    <p>We are trying to make privacy, strong cryptography and authentication the Internet norm.</p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://www.tribler.org/img/image0.png</image>
+      <caption>Screenshot of the Tribler UI</caption>
+    </screenshot>
+    <screenshot>
+      <image>https://www.tribler.org/img/image3.png</image>
+      <caption>Screenshot of the Tribler UI</caption>
+    </screenshot>
+    <screenshot>
+      <image>https://www.tribler.org/img/image1.png</image>
+      <caption>Screenshot of the Tribler UI</caption>
+    </screenshot>
+    <screenshot>
+      <image>https://www.tribler.org/img/image4.png</image>
+      <caption>Screenshot of the Tribler UI</caption>
+    </screenshot>
+    <screenshot>
+      <image>https://www.tribler.org/img/image9.png</image>
+      <caption>Screenshot of the Tribler UI</caption>
+    </screenshot>
+    <screenshot>
+      <image>https://www.tribler.org/img/image10.png</image>
+      <caption>Screenshot of the Tribler UI</caption>
+    </screenshot>
+  </screenshots>
+  <launchable type="desktop-id">org.tribler.Tribler.desktop</launchable>
+  <content_rating type="oars-1.1">
+    <content_attribute id="social-info">mild</content_attribute>
+  </content_rating>
+  <releases>
+    <release version="7.8.0" date="2021-02-15"/>
```

Comment:
We refrain from hard-coding versions in our files. Please have a look at [this script](https://github.com/Tribler/tribler/blob/main/build/update_version_from_git.py) that is executed before packaging Tribler 👍

Reply:
@overflw this is a way to satisfy this request for changes:

1. Remove the last three lines from this file (line 47, line 48 and line 49) and leave the last line blank.

2. Paste the following code into the `__main__` section of [that script](https://github.com/Tribler/tribler/blob/main/build/update_version_from_git.py):

```python
logger.info('Writing AppStream version.')
with open(path.join('build', 'debian', 'tribler', 'usr', 'share', 'metainfo',
                    'org.tribler.Tribler.metainfo.xml'), 'a') as f:
    f.write(f'    <release version="{version_id}" date="{build_date}"/>'
            f'\n  </releases>\n</component>')
```

Reply:
Hey (:  thanks for the hints! I wrote a xml parsing based addendum to `update_version_from_git.py`:
```python
    logger.info('Writing AppStream version info.')
    releaseDate = time.strftime("%Y-%m-%d", time.localtime())
    attrib = {'version': f'{version_id}', 'date':f'{releaseDate}'}

    tree = xml.parse(path.join('build', 'debian', 'tribler', 'usr', 'share', 'metainfo',
                        'org.tribler.Tribler.metainfo.xml'))
    xmlRoot = tree.getroot()
    releases = xmlRoot.find('releases')
    release = xml.SubElement(releases, 'release', attrib)
    tree.write(path.join('build', 'debian', 'tribler', 'usr', 'share', 'metainfo',
                        'org.tribler.Tribler.metainfo.xml'))
```